framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -718,12 +750,41 @@ protected void addInformationFromPreconditions(S info,
 
         S info = in.getRegularStore();
         V rhsValue = in.getValueOfSubNode(rhs);
+        Receiver expr = FlowExpressions.internalReprOf(
+                analysis.getTypeFactory(), lhs);
+        if (inferSignatures && expr instanceof FieldAccess &&
+                !analysis.checker.shouldSuppressWarnings(n.getTree(), null) &&
```

Comment:
I would extract the `shouldSuppressWarnings` checks into a helper method. It appears in three duplicates and one partial duplicate.